seqr/utils/search/elasticsearch/es_utils.py

```diff
@@ -248,3 +250,69 @@ def _get_es_indices(client):
 
     return indices, seqr_index_projects
 
+
+def get_es_variants_for_variant_ids(families, variant_ids, user, dataset_type=None, return_all_queried_families=False):
+    variants = EsSearch(
+        families, user=user, return_all_queried_families=return_all_queried_families,
+    ).filter_by_variant_ids(variant_ids)
+    if dataset_type:
+        variants = variants.update_dataset_type(dataset_type)
+    return variants.search(num_results=len(variant_ids))
+
+
+def get_es_variants(families, search, user, previous_search_results, sort=None, page=None, num_results=None,
+                    gene_agg=False, skip_genotype_filter=False):
+    es_search_cls = EsGeneAggSearch if gene_agg else EsSearch
+
+    es_search = es_search_cls(
+        families,
+        previous_search_results=previous_search_results,
+        user=user,
+        sort=sort,
+    )
+
+    es_search.filter_variants(
+        inheritance=search.get('inheritance'), frequencies=search.get('freqs'), pathogenicity=search.get('pathogenicity'),
+        annotations=search.get('annotations'), annotations_secondary=search.get('annotations_secondary'),
+        in_silico=search.get('in_silico'), quality_filter=search.get('qualityFilter'),
+        custom_query=search.get('customQuery'), locus=search.get('locus'), skip_genotype_filter=skip_genotype_filter,
+        **search.get('parsedLocus')
+
+    )
+
+    return es_search.search(page=page, num_results=num_results)
+
+
+def process_es_previously_loaded_results(previous_search_results, start_index, end_index):
+    grouped_results = previous_search_results.get('grouped_results')
+    results = None
+    if grouped_results:
+        results = get_compound_het_page(grouped_results, start_index, end_index)
+
+    return results
+
+
+def process_es_previously_loaded_gene_aggs(previous_search_results):
```

Comment:
This function does little with the `ES`. You may want to remove the `es` from its name and move it to the search util file.

Reply:
It is ES specific, only ES searches will generate this weird partial cache field, the hail backend will not, so I broke the caching apart into the caching that will be shared and the caching that will be ES only